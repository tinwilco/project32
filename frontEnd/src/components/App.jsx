import React, { useState } from "react";

import EmojiButton from "./EmojiButton/EmojiButton";
import styles from "./App.module.css";

function App() {
  const handleButtonPress = label => {
    updateSelectedMood(label);
  };

  const emojiButtonArray = [
    { label: "ðŸ¤ " },
    { label: "ðŸ¤‘" },
    { label: "ðŸ¤§" },
    { label: "ðŸ¥³" },
    { label: "ðŸ˜­" }
  ];

  const [selectedMood, updateSelectedMood] = useState("");

  return (
    <div className={styles.App}>
      <header className={styles.App_header}>
        <h2>How are you feeling today?</h2>
        <div className={styles.App_Button_Container}>
          {emojiButtonArray.map(button => {
            return (
              <EmojiButton
                label={button.label}
                key={`emoji-button-${button.label}`}
                handleButtonPress={() => handleButtonPress(button.label)}
                selected={selectedMood === button.label}
              />
            );
          })}
        </div>
      </header>
    </div>
  );
}

export default App;
