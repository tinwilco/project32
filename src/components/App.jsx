import { useState } from "react";
import { Button } from "reactstrap";

import logo from "../assets/logo.png";
import ColleagueInterface from "./ColleagueInterface/ColleagueInterface";
import ManagerInterface from "./ManagerInterface/ManagerInterface";

import styles from "./App.module.css";

function App() {
  const [isManager, updateIsManager] = useState(false);

  return (
    <div className={styles.App}>
      <div className={styles.App_background}>
        <div className={styles.App_container}>
          <img
            className={styles.logo}
            src={logo}
            alt="Hedwig Logo"
            title="Hedwig Logo"
          />
          {!isManager ? <ColleagueInterface /> : <ManagerInterface />}
        </div>
        <Button
          color="link"
          onClick={() => updateIsManager(!isManager)}
          className={styles.SwitchModeButton}
        >
          {isManager
            ? "Return to Mood Entry"
            : "Manager? Click here to see team moods!"}
        </Button>
      </div>
    </div>
  );
}

export default App;
