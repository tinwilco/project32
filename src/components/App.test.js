import React from "react";
import { render } from "@testing-library/react";
import MockDate from "mockdate";

import App from "./App";

describe("App component", () => {
  beforeAll(() => {
    MockDate.set("2020-01-01");
  });
  afterAll(() => {
    MockDate.reset();
  });
  it("renders to match the snapshot", () => {
    const subject = render(<App />);
    expect(subject.baseElement).toMatchSnapshot();
  });
});
