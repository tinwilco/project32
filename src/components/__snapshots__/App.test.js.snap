// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders to match the snapshot 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <div
        class="App_background"
      >
        <div
          class="App_container"
        >
          <img
            alt="Hedwig Logo"
            class="logo"
            src="logo.png"
            title="Hedwig Logo"
          />
          <h2>
            Hello, How are you feeling today?
          </h2>
          <h3>
            Wednesday 01/01/2020
          </h3>
          <p
            class="helpText"
          >
            Please select from the options below
          </p>
          <div
            class="App_Button_Container"
          >
            <button
              class="EmojiButton"
              type="button"
            >
              <span
                aria-label="miserable"
                role="img"
              >
                😢
              </span>
            </button>
            <button
              class="EmojiButton"
              type="button"
            >
              <span
                aria-label="sad"
                role="img"
              >
                🙁
              </span>
            </button>
            <button
              class="EmojiButton"
              type="button"
            >
              <span
                aria-label="neutral"
                role="img"
              >
                😑
              </span>
            </button>
            <button
              class="EmojiButton"
              type="button"
            >
              <span
                aria-label="happy"
                role="img"
              >
                🙂
              </span>
            </button>
            <button
              class="EmojiButton"
              type="button"
            >
              <span
                aria-label="ecstatic"
                role="img"
              >
                🥳
              </span>
            </button>
          </div>
          <label
            class=""
            for="usernameInputTextBox"
          >
            Who are you?
          </label>
          <input
            class="UsernameEntry form-control-lg form-control"
            id="usernameInputTextBox"
            placeholder="Enter your name"
            type="text"
            value=""
          />
          <button
            class="submitButton btn btn-primary btn-lg disabled"
            disabled=""
            type="button"
          >
            Submit
          </button>
        </div>
        <button
          class="SwitchModeButton btn btn-link"
          type="button"
        >
          Manager? Click here to see team moods!
        </button>
      </div>
    </div>
  </div>
</body>
`;
